src/database/alert_dao.rs

```diff
@@ -0,0 +1,536 @@
+use super::database::Database;
+use crate::{
+    errors::ServicesError,
+    expect_log,
+    reporting::{
+        public_report::{CoughSeverity, FeverSeverity, PublicReport},
+        symptom_inputs::UserInput,
+    },
+    reports_interval,
+    reports_update::reports_updater::Alert,
+};
+use log::*;
+use reports_interval::UnixTime;
+use rusqlite::{params, Row, NO_PARAMS};
+use std::sync::Arc;
+
+pub trait AlertDao {
+    fn all(&self) -> Result<Vec<Alert>, ServicesError>;
+    fn save(&self, alerts: Vec<Alert>) -> Result<(), ServicesError>;
+    fn delete(&self, id: String) -> Result<(), ServicesError>;
+}
+
+pub struct AlertDaoImpl {
+    db: Arc<Database>,
+}
+
+impl AlertDaoImpl {
+    pub fn new(db: Arc<Database>) -> AlertDaoImpl {
+        Self::create_table_if_not_exists(&db);
+        AlertDaoImpl { db }
+    }
+
+    fn create_table_if_not_exists(db: &Arc<Database>) {
+        // TODO use blob for tcn? https://docs.rs/rusqlite/0.23.1/rusqlite/blob/index.html
+        // TODO ideally FFI should send byte arrays too
+        let res = db.execute_sql(
+            "create table if not exists alert(
```

Comment:
Convention is writing sql queries is that sql keywords are capitalized - helps with readability:
```
"CREATE TABLE IF NOT EXISTS alert(
id TEXT PRIMARY KEY,...
```
Can be made into a "good first issue".

Reply:
I'm aware of this convention, but AFAIK it's not enforced and I personally don't see a need for it, as it's always pretty obvious what the keywords are. Open to discuss with the team, though, as it's just my personal preference.

These free text queries are anyway temporary, as there's a lot of duplication and it's error prone. We could use an ORM or some sort of DSL to make it safer.